mlops/nyc_taxi/src/mlops_pipeline.py

```diff
@@ -78,261 +71,101 @@ def nyc_taxi_data_regression(pipeline_job_input, model_name, build_reference):
-    pipeline_job.settings.default_datastore = "workspaceblobstore"
+        Args:
+            ml_client: The Azure ML client to use for retrieving data assets and components.
 
-    return pipeline_job
-
-
-def execute_pipeline(
-    subscription_id: str,
-    resource_group_name: str,
-    workspace_name: str,
-    experiment_name: str,
-    pipeline_job: pipeline,
-    wait_for_completion: str,
-    output_file: str,
-):
-    """
-    Execute a pipeline job in Azure Machine Learning service.
-
-    Args:
-        subscription_id (str): The Azure subscription ID.
-        resource_group_name (str): The name of the resource group.
-        workspace_name (str): The name of the Azure Machine Learning workspace.
-        experiment_name (str): The name of the experiment.
-        pipeline_job (pipeline): The pipeline job to be executed.
-        wait_for_completion (str): "True" or "False" indicates whether to wait for the job to complete.
-        output_file (str): The path to the output file where the job name will be written.
-
-    Raises:
-        Exception: If the job fails to complete.
-
-    Returns:
-        None
-    """
-    try:
-        client = MLClient(
-            DefaultAzureCredential(),
-            subscription_id=subscription_id,
-            resource_group_name=resource_group_name,
-            workspace_name=workspace_name,
+        Returns:
+            pipeline_job: The constructed pipeline job components.
+        """
+        registered_data_asset = ml_client.data.get(
+            name=self.dataset_name, label="latest"
         )
 
-        pipeline_job = client.jobs.create_or_update(
-            pipeline_job, experiment_name=experiment_name
+        parent_dir = os.path.join(os.getcwd(), "mlops/nyc_taxi/components")
+
+        prepare_data = load_component(source=parent_dir + "/prep.yml")
+        transform_data = load_component(source=parent_dir + "/transform.yml")
+        train_model = load_component(source=parent_dir + "/train.yml")
+        predict_result = load_component(source=parent_dir + "/predict.yml")
+        score_data = load_component(source=parent_dir + "/score.yml")
+        register_model = load_component(source=parent_dir + "/register.yml")
+
+        # Set the environment name to custom environment using name and version number
+        prepare_data.environment = self.environment_name
+        transform_data.environment = self.environment_name
+        train_model.environment = self.environment_name
+        predict_result.environment = self.environment_name
+        score_data.environment = self.environment_name
+        register_model.environment = self.environment_name
+
+        gl_pipeline_components.append(prepare_data)
+        gl_pipeline_components.append(transform_data)
+        gl_pipeline_components.append(train_model)
+        gl_pipeline_components.append(predict_result)
+        gl_pipeline_components.append(score_data)
+        gl_pipeline_components.append(register_model)
```

Comment:
similar recommendation to a previous comment for making this concise

Reply:
updated